src/retrieval/get_latest_for_entry.rs

```diff
@@ -31,50 +31,79 @@ impl GetLatestEntry {
         &self,
         entry_hash: EntryHash,
         get_options: GetOptions,
-    ) -> ExternResult<OptionEntryAndHash<T>> {
-        // First, make sure we DO have the latest header_hash address
-        let maybe_latest_header_hash = match get_details(entry_hash.clone(), get_options.clone())? {
+    ) -> ExternResult<Option<WireElement<T>>> {
+        match get_details(entry_hash.clone(), get_options.clone())? {
             Some(Details::Entry(details)) => match details.entry_dht_status {
-                metadata::EntryDhtStatus::Live => match details.updates.len() {
+                metadata::EntryDhtStatus::Live => {
+                    let created_at = details.headers.first().unwrap().header().timestamp(); // TODO: better error handling (avoid unwrap)
```

Comment:
yeah this should be ok for now... 

maybe assign `details.headers.first().unwrap()` to a variable though and then you can re use it below on line 43